### PR TITLE
Try/catch for Docker client

### PR DESCRIPTION
**Pull request by christianversloot**
## Describe changes
When your Docker daemon is not running in a Docker based stack, users get somewhat obscure errors like `DockerException: Error while fetching server API version: ('Connection aborted.', FileNotFoundError(2, 'No such file or directory'))`.

This PR wraps all `DockerClient.from_env()` calls where no try/catch construct is present with one, raising a more user-friendly error as to what is wrong and could be done to help resolve the issue.

## Pre-requisites
Please ensure you have done the following:
- [x] I have read the **CONTRIBUTING.md** document.
- [x] If my change requires a change to docs, I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [x] I have based my new branch on `develop` and the open PR is targeting `develop`. If your branch wasn't based on develop read [Contribution guide on rebasing branch to develop](https://github.com/zenml-io/zenml/blob/main/CONTRIBUTING.md#-pull-requests-rebase-your-branch-on-develop).
- [x] If my changes require changes to the dashboard, these changes are communicated/requested.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Other (add details above)

## Summary by CodeRabbit

- **New Features**
	- Not applicable.
- **Bug Fixes**
	- Improved error handling across various components when creating a Docker client fails, providing clearer error messages to help diagnose issues with the Docker environment.
- **Documentation**
	- Not applicable.
- **Refactor**
	- Enhanced exception handling for Docker client creation to ensure consistent error reporting.
- **Style**
	- Not applicable.
- **Tests**
	- Not applicable.
- **Chores**
	- Not applicable.
- **Revert**
	- Not applicable.

### CODE REVIEW

**Comment by strickvl** [comment]
@coderabbitai review

File: src/zenml/utils/docker_utils.py [hunk 1]
```diff
@@ -321,7 +344,12 @@ def is_local_image(image_name: str) -> bool:
     Returns:
         `True` if the image was pulled from a registry, `False` otherwise.
     """
-    docker_client = DockerClient.from_env()
+    try:
+        docker_client = DockerClient.from_env()
+    except DockerException as e:
+        raise RuntimeError(
+            "Could not create a Docker client from the environment. Is your Docker daemon running?"
+        ) from e
```
Comment (avishniakov) [inline src/zenml/utils/docker_utils.py:352]: I suggest you wrap this logic into utility method in `src/zenml/utils/docker_utils.py`, something like:
```python
def _try_get_docker_client_from_env():
    try:
        return DockerClient.from_env()
    except DockerException as e:
        raise RuntimeError(
            "Could not create a Docker client from the environment. Is your Docker daemon running?"
        ) from e
```
and use it in all other place to safe some LOC.

**Review by avishniakov — CHANGES_REQUESTED** [review]
Hey, thanks for contributing! I left one suggestion. Also the docstring are impacted now - my suggestion will also simplify resolution for this issue.

**Review by avishniakov — APPROVED** [review]
Good to go, once CI passes

**Comment by christianversloot** [comment]
thanks for merging @strickvl !

**Pull request merged**